CLI: unexpected error during provisioning

Reported by diegozuluaga@google.com:

Error running provision on alpha-3:
```
apigee-istio provision -o testmyapi -e test -u **** -p ***** > samples/apigee/handler.yaml

panic: runtime error: invalid memory address or nil pointer dereference[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7bba0e]
```
On Apigee: kvm(istio) and api proxy (istio-auth) are there. API Proxy didn't deploy(red triangle).

Output with verbose:
```
HTTP/1.1 404 Not Found
Content-Length: 167
Connection: keep-alive
Content-Type: application/json
Date: Wed, 18 Jul 2018 21:20:39 GMT
Server: Apigee LB

{
  "code" : "messaging.config.beans.ApplicationDoesNotExist",
  "message" : "APIProxy named istio-auth does not exist in organization testmyapi",
  "contexts" : [ ]
}

    "available": 0,
    "expiryTime": 0,
    "timestamp": 1531948912314
}

verifying customer proxy...
GET /istio-auth/certs HTTP/1.1^M
Host: testmyapi-test.apigee.net^M
User-Agent: Go-http-client/1.1^M
Accept-Encoding: gzip
```

When provisioning fails as above, it must fail with a good error message.

--- a/apigee-istio/cmd/provision/provision.go
+++ b/apigee-istio/cmd/provision/provision.go
@@ -539,6 +539,9 @@ func (p *provision) importAndDeployProxy(name string, proxy *apigee.Proxy, oldRe
 		printf("undeploying proxy %s revision %d on env %s...",
 			name, oldRev, p.Env)
 		_, resp, err = p.Client.Proxies.Undeploy(name, p.Env, *oldRev)
+		if err != nil {
+			return errors.Wrapf(err, "error undeploying proxy %s", name)
+		}
 	}
 
 	printf("deploying proxy %s revision %d to env %s...", name, newRev, p.Env)

